Index: 03_picnic/test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- 03_picnic/test.py	(revision 9f159863721610b383ecb6cf82eaa137b9680038)
+++ 03_picnic/test.py	(date 1598647109382)
@@ -66,3 +66,40 @@
     out = getoutput(f'{prg} {arg} --sorted')
     expected = ('You are bringing apples, bananas, cherries, and dates.')
     assert out.strip() == expected
+
+# --------------------------------------------------
+def test_more_than_two_without_oxford_comma():
+    """user can choose not to print oxford comma """
+
+    arg = 'bananas apples dates'
+    out = getoutput(f'{prg} {arg} --comma')
+    expected = ('You are bringing bananas, apples and dates.')
+    assert out.strip() == expected
+
+# --------------------------------------------------
+def test_more_than_two_sorted_without_oxford_comma():
+    """user can choose not to print oxford comma """
+
+    arg = 'bananas apples dates'
+    out = getoutput(f'{prg} {arg} --comma --sorted')
+    expected = ('You are bringing apples, bananas and dates.')
+    assert out.strip() == expected
+
+# --------------------------------------------------
+def test_more_than_two_separator():
+    """more than two items with different separator"""
+
+    arg = '"potato chips" coleslaw cupcakes "French silk pie"'
+    out = getoutput(f"{prg} {arg} --sep ';' ")
+    expected = ('You are bringing potato chips; coleslaw; '
+                'cupcakes; and French silk pie.')
+    assert out.strip() == expected
+
+# --------------------------------------------------
+def test_more_than_two_sorted_separator():
+    """more than two items, sorted, with different separator"""
+
+    arg = 'bananas apples dates'
+    out = getoutput(f"{prg} {arg} --sorted --sep ';' ")
+    expected = ('You are bringing apples; bananas; and dates.')
+    assert out.strip() == expected
\ No newline at end of file
